### ISSUE TITLE
Unexpected behavior of CommandAction("core show channels");

### ISSUE THREAD

**Issue by BastardFat** [AsterNET/AsterNET#81]
I'm using this code to get actual channels:
`CommandResponse response = manager.SendAction(new CommandAction("core show channels concise")) as CommandResponse;`

Usually the last line of the response I get "`--END COMMAND--`", but sometimes this does not happen.  How can I protect myself from error when the command response is received not fully?
![image](https://cloud.githubusercontent.com/assets/23219786/21304022/7c234134-c5d3-11e6-856d-7ddaa6ecd215.png)

**Comment by skrusty** [comment]
Is Asterisk not sending the final line, or are we (AsterNET) not processing it?

**Comment by BastardFat** [comment]
Actualy I dont know. I use AsterNet as Nuget (without debug info). Earlier, I wrote a small console application wrapper, which is sent to the CLI command and outputting all of its response to the console. For a long time use, it has never occurred such a problem - "--END KOMMAND--" always appears. I guess that the problem may be due to the fact that I'm trying to use this command for real-time monitoring, and every second (on a timer) to perform it

**Comment by skrusty** [comment]
Good tip, check with Wireshark. AsterNET won't handle that being missing well at all, it's expected. If you find you're not getting it in the TCP traces, then I would speak to someone in #asterisk on irc or the usergroup

**Comment by BastardFat** [comment]
Thank you. I'll check the traffic, and if I find out potential cause of the error, I will write here

**Comment by BastardFat** [comment]
![image](https://cloud.githubusercontent.com/assets/23219786/21308599/b9c89c88-c5eb-11e6-9b67-f58b4bc58846.png)
Asterisk is always says "--END COMMAND--" in the last line.
Can you try this code?
![image](https://cloud.githubusercontent.com/assets/23219786/21308865/e93ba61c-c5ec-11e6-86b4-2344151aeedd.png)
Sometimes he makes the following output:
![image](https://cloud.githubusercontent.com/assets/23219786/21308948/38ec8e60-c5ed-11e6-8bd3-b55d08315426.png)

**Comment by skrusty** [comment]
I suggest, get the code version, compile and test to see if you still get the same behaviour. If you do, you might be able to track it down quite easily, otherwise it may be an issue that's already been addressed. 

Right now I don't have time to review and test myself, sorry.

**Comment by jwmoffat** [comment]
I can duplicate the issue using @BastardFat's sample code for both the latest NuGet package (`1.0.1`) and the current master code (commit 714482d9).   My test code was using `Asterisk 11.2.1`.

Out of 1000 runs, I would get an invalid item 10 times or less, but I still get it every batch run.

I did spot something interesting, though.  I adjusted the test code to count the frequency of invalid results and set a break-point to see what the last result was if it happened.  This is what I saw:

![image](https://cloud.githubusercontent.com/assets/7852605/21461398/1aa59336-c906-11e6-9b11-444e0f2277c6.png)

Yes, I know I misspelled the command, but what's interesting is that the list `t.Result` appears to be still updating, on occasion, after the `SendAction` returns.

I did a couple of test runs of putting a 100ms delay prior to querying the result list and that appears to "resolve" the issue, though it's probably not ideal.

I presume the `SendAction`, in this case, shouldn't be returning until the result list is fully populated?

**Comment by skrusty** [comment]
Can those with this issue please review the patch submitted to fix this. Feedback on pull request welcome.

**Comment by jwmoffat** [comment]
Looking at the code before/after, it makes sense why this resolves it and why I was seeing the list populate after the fact.  Plus I don't see any issues with the changes that were done.

From a code test, I ran the patch against two 1000-set runs and I didn't see the issue.  Switching back to the current master I was able to get the issue right away. 

I'd say it's good, at least from my perspective.  👍
